File: Problem Sets/pset2/readability.c
#include <cs50.h>
#include <stdio.h>
#include <string.h>
#include <ctype.h>
#include <math.h>

int main(void)
{
    //Input string
    string text = get_string("text: ");

    //counters
    float words = 0;
    float sentences = 0;
    float letters = 0;

    int len = strlen(text);

    //Loops through all the letters in imputted string to calculate n° of Sentences/Words/Letters
    for (int a = 0; a < len; a++)
    {
        if (isalpha(text[a]) )
        {
            letters++;
        }

        if (text[a] == '.' || text[a] == '?' || text[a] == '!')
        {
            sentences++;
            words++;
        }

        if ( (isspace(text[a]) || text[a] == ',' || text[a] == ':' || text[a] == ';') && isalpha(text[a-1]) )
        {
            words++;
        }

    }


    //Calculate L/S/ColemanLiau's formula
    float L = (letters / words) * 100;
    float S = (sentences / words) * 100;
    int colemanliau = round( (0.0588 * L) - (0.296 * S) - 15.8 );

    if (colemanliau > 16)
    {
        printf ("Grade 16+\n");
        return false;
    }

    if (colemanliau < 1)
    {
        printf("Before Grade 1\n");
        return false;
    }


    printf ("Grade %i\n", colemanliau);












}

File: Problem Sets/pset6/readability/readability.py
from cs50 import get_string

frase = get_string ("Inserisci il testo: ")

#sum letters
letters = 0.00
#sum sentences
sentences = 0.00
#number of words
words = 0.00

#Iterate throught frase for counting words/letters/sentences
for letter in range(len(frase)):

    #if char is letter -> letter ++
    if frase[letter].isalpha():
        letters += 1.00

    #if char is . or ! or ? (end of a sentence) -> sentence and word ++ as is also a end for a word
    if frase[letter] == "." or frase[letter] == "!" or frase[letter] == "?":
        sentences +=  1.00
        words += 1.00

    #if char is space and last char was letter -> end of a word, so word++
    if frase[letter].isspace() and (frase[letter - 1].isalpha() or frase[letter - 1] == "," or frase[letter - 1] == ":" or frase[letter - 1] == ";") :
        words +=  1.00


#Calculate the L S and Coleman Liau Index
letters = (letters / words) * 100.00
sentences = (sentences / words) * 100.00
colemanliau = (0.0588 * letters) - (0.296 * sentences) - 15.80



#round to nearest integer because check50 is freaking unhappy
colemanliau = round(colemanliau)

#Print grade: above 16/below 1/Actual Grade
if colemanliau > 16:
    print ("Grade 16+")


elif colemanliau < 1:
    print ("Before Grade 1")

else:
    print (f"Grade {colemanliau}")